keps/sig-cli/20191008-kubectl-events.md

```diff
@@ -0,0 +1,152 @@
+      --no-headers=false: When using the default or custom-column output format, don't print headers (default print
+headers).
+  -o, --output='': Output format. One of:
+json|yaml|wide|name|custom-columns=...|custom-columns-file=...|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=...
+See custom columns [http://kubernetes.io/docs/user-guide/kubectl-overview/#custom-columns], golang template
+[http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template
+[http://kubernetes.io/docs/user-guide/jsonpath].
+      --output-watch-events=false: Output watch event objects when --watch or --watch-only is used. Existing objects are
+output as initial ADDED events.
+      --raw='': Raw URI to request from the server.  Uses the transport specified by the kubeconfig file.
+  -l, --selector='': Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)
+      --server-print=true: If true, have the server return the appropriate table output. Supports extension APIs and CRDs.
+      --show-kind=false: If present, list the resource type for the requested object(s).
+      --show-labels=false: When printing, show all labels as the last column (default hide labels column)
+      --sort-by='': If non-empty, sort list types using this field specification.  The field specification is expressed
+as a JSONPath expression (e.g. '{.metadata.name}'). The field in the API resource specified by this JSONPath expression
+must be an integer or a string.
+  -w, --watch=false: After listing/getting the requested object, watch for changes. Uninitialized objects are excluded
+if no object name is provided.
+      --watch-only=false: Watch for changes to the requested object(s), without listing/getting first.
+
+  ```
+  
+
+#### Long standing issues
+
+Provide a default sort order for the `events` rather than the random order: 
+
+- kubectl get events doesnt sort events by last seen time [kubernetes/kubernetes#29838](https://github.com/kubernetes/kubernetes/issues/29838)
+
+### Goals
+
+- Add an experimental `events` sub-command under the kubectl and move the existing `events` command from `kubectl get` 
+- This new command will not be dependent on `kubectl get`
+
+## Proposal
+
+Have a independent *events* command which can perform all the existing tasks that the current `kubectl get events` 
+command is performing
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+After the successful adoption of the command, the `events` command under the `kubectl get` can be removed. 
+
+- Create the *kubectl events* sub command (Work in Progress)
+
+
+## Graduation Criteria
+
+- The `events` command has been added to `kubectl` 
+- Publish `events` as a subcommand of kubectl.
+- Remove the existing `kubectl` from under the `kubectl get` 
+
+
+## Alternatives
+
+1. Modify the existing code and provide a default sort order for the events. [PR](https://github.com/kubernetes/kubernetes/pull/82898)
+
```

Comment:
Could `kubectl events` work / ship as a [plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/)?

Reply:
Not really. It will be a part of `kubectl` as the other sub-commands are.

Reply:
@hpandeycodeit the point I'm making is that a subcommand plugin is a viable _alternative_, not that the KEP should focus on the plugin approach.

Does that make sense?

Reply:
It can be a plugin, although sig-cli things it's worthy to have this as a built-in commands, since currently we don't have good mechanism for viewing events and they are important piece of feedback you can gather from the platform.